app/commands/git/sync_concept_exercise.rb

```diff
@@ -88,6 +88,11 @@ def exercise_config
       head_git_track.concept_exercises[exercise_index]
     end
 
+    memoize
+    def exercise_config_taught_prerequisites
```

Comment:
```suggestion
    def exercise_config_prerequisites
```

I don't think we need to have these as "taught" they're effectively just the "valid" data.